python/lsst/daf/persistence/butler.py

```diff
@@ -136,8 +139,36 @@ def __init__(self, root, mapper=None, **mapperArgs):
         self.log = pexLog.Log(pexLog.Log.getDefaultLog(),
                 "daf.persistence.butler")
 
-    def getKeys(self, datasetType=None, level=None):
+    def defineAlias(self, alias, datasetType):
+        """Register an alias that will be used in place datasetType keyword.
+
+        @param alias (str) the alias keyword. it may start with @ or not. It may not contain @ except as the
+                           first character.
+        @param datasetType (str) the keyword that will be used when @alias is passed into datasetType. it may
+                                 not contain '@'
+        """
+
+        #verify formatting of alias:
+        # it can have '@' as the first character (if not it's okay, we will add it) or not at all.
+        atLoc = alias.rfind('@')
+        if atLoc is -1:
+            alias = "@%s" %(alias)
+        elif atLoc > 0:
+            raise RuntimeError("Badly formatted alias string:%s" %(alias))
```

Comment:
I'd put a space after the colon.  It's slightly more accurate to use a comma after `alias` to indicate that this is a one-item tuple.  But using `.format()` instead of `%` might be even better.